lib/engine/ability/token.rb

```diff
@@ -5,13 +5,14 @@
 module Engine
   module Ability
     class Token < Base
-      attr_reader :hexes, :price, :teleport_price, :extra
+      attr_reader :hexes, :price, :teleport_price, :extra, :from_owner
 
-      def setup(hexes:, price:, teleport_price: nil, extra: nil)
+      def setup(hexes:, price:, teleport_price: nil, extra: nil, from_owner: false)
```

Comment:
this can just be nil because you already do or false

Reply:
Fixed